virttest/qemu_devices/qcontainer.py

```diff
@@ -2940,7 +2940,7 @@ def thread_context_define_by_params(self, params, name):
         prefix = 'vm_thread_context_'
         for key in params:
             if key.startswith(prefix):
-                new_key = key.rsplit(prefix)[1]
+                new_key = key.rsplit(prefix)[1].rsplit('_')[0]
```

Comment:
Hi @mcasquer , would you like to elaborate more on the issue you observed? My understanding is that, as we've already used the per object params for populating thread contexts (via invoking `params.object_params` within the VM main routine of avocado-vt), it seems that we would not encounter into such a case, so did I miss something?

https://github.com/avocado-framework/avocado-vt/blob/802869f8dbbd350afae369de5c8db41462857229/virttest/qemu_vm.py#L1280

Reply:
ok, then I guess a quick fix for this issue would be:

```patch
-            if key.startswith(prefix):
+            if key.startswith(prefix) and not key.endswith(f"_{name}"):
```

Furthermore, regarding a regular fix I think we could introduce something like the following to resolve all the similar issues for any other cases.
```patch
diff --git a/virttest/utils_params.py b/virttest/utils_params.py
index 74619afd..ce7f11a0 100644
--- a/virttest/utils_params.py
+++ b/virttest/utils_params.py
@@ -57,7 +57,7 @@ class Params(IterableUserDict):
         del lst
         return objs
 
-    def object_params(self, obj_name):
+    def object_params(self, obj_name, drop_name_key=False):
         """
         Return a dict-like object containing the parameters of an individual
         object.
@@ -78,6 +78,8 @@ class Params(IterableUserDict):
             if key.endswith(suffix):
                 new_key = key.split(suffix)[0]
                 new_dict[new_key] = new_dict[key]
+                if drop_name_key:
+                    del new_dict[key]
         return new_dict
 
     def object_counts(self, count_key, base_name):
```

Reply:
@luckyh IIUC this is going to delete the param according to the received name, but keep in mind that the dict contains **all the params** so i.e. for the name 'tc1' deletes the `vm_thread_context_node-affinity_tc1` but not `vm_thread_context_node-affinity_tc2`

Reply:
Then the condition above will pass when evaluating `key = vm_thread_context_node-affinity_tc2` for `name = tc1`

Reply:
@luckyh any update about my latest commets? Thanks !

Reply:
@mcasquer good catch! And apologize for not thought out when last time I reply this thread.

Since the `cartesian_config` doesn't support the identification of object names at grammar level, we could not get the expected target param by merely using `.rsplit('_')[0]`, due to `_` is also a valid character for naming objects (e.g. `1st_tc`) and that could lead wrong splitting.

A "safe" way to fix the issue is that we could retrieve the tc params in a static manner, where the drawback is that the code has to be modified when everytime a new option is going to be supported.
```python
tc_params["foo"] = params["vm_thread_context_foo"]
tc_params["bar"] = params["vm_thread_context_bar"]
...
```
or do you have any other idea?

Reply:
Updated!